aiven/client/connection_info/_utils.py

```diff
@@ -1,22 +1,27 @@
 # Copyright (c) 2021 Aiven, Helsinki, Finland. https://aiven.io/
 from .common import ConnectionInfoError
 from aiven.client.common import UNDEFINED
-from typing import Mapping, Optional
+from typing import Any, Mapping, Optional, Sequence, TypeVar
 
 import ipaddress
 import urllib.parse
 
 
-def find_component(items, **filters):
+T = TypeVar("T")
+
+
+def find_component(items: Sequence[Mapping[str, T]], **filters: str) -> Mapping[str, T]:
     for item in items:
         if all(key in item and item[key] == value for key, value in filters.items() if value is not UNDEFINED):
             return item
     msg_filters = ", ".join(f"{field}={value}" for field, value in filters.items())
     raise ConnectionInfoError(f"Could not find connection information with filters {msg_filters}")
 
 
-def find_user(service, username):
-    for user in service["users"]:
+def find_user(service: Mapping[str, T], username: str) -> Mapping[str, T]:
```

Comment:
I think type variable is wrong and confusing here. Type for `service` argument looks like `Mapping[str, Mapping[str, Mapping[str, str]]]` and the result is probably `Mapping[str, str]`. So if type for values in the result isn't always `str` then probably `service` argument should be `Mapping[str, Mapping[str, Mapping[str, T]]]` instead.

Reply:
I think should not assume more than this:
```
def find_user(service: Mapping[str, Any], username: str) -> Mapping[str, Any]:
    users = service["users"]
    assert isinstance(users, Mapping)
```
since it's arbitrary depth JSON which can contain integers, nulls, etc.